### PR TITLE
mps-cli-gradle-plugin: COI based on models

### PR DESCRIPTION
**Pull request by kolmar**
If no modules or models were removed, then modules in the cone of influence are now computed based on the imports of modified models, instead of dependencies on modified modules. The finer granularity allows to have a smaller cone of influence.

Additional refactorings:
* Combined `ConeOfInfluenceComputerTest_01` and `ConeOfInfluenceComputerTest_02` into `ConeOfInfluenceComputerTest` with several test cases. Added additional tests cases to it, to test the new functionality.
* `computeConeOfInfluence` task no longer depends on `buildModuleDependencies`. Depending on which files were modified in the repository, we may need to use either module dependencies or model dependencies. If we depend on both `buildModuleDependencies` and `buildModelDependencies`, then each computes its own `SRepository`, where the same models/modules have different object IDs.
* Common code to compute to compute model/module dependencies moved into a new class `EntityDependenciesBuilder`. It's used by all 3 tasks `buildModelDependencies`, `buildModuleDependencies` and `computeConeOfInfluence`
* `Filesystem2SSolutionBridge` now finds both modules and models parts of which were modified and tracks if any models or modules were deleted.
    * It also now stops at the repo directory. It doesn't go further into the ancestor directories of git repository.
* Modified files are now linked to models/modules in the repository through `path.toRealPath()` method

### CODE REVIEW

File: mps-cli-gradle-plugin/plugin/src/main/groovy/org/mps_cli/cone_of_influence/Filesystem2SSolutionBridge.groovy [hunk 1]
```diff
@@ -8,29 +13,83 @@ import static groovy.io.FileType.FILES
...
-                def myFile = gitRepoLocation.resolve(it).parent
-                def affectedModuleFound = false
-                while (myFile != null) {
-                    if (Files.exists(myFile)) {
-                        myFile.traverse(type: FILES, maxDepth: 0, nameFilter: filterSolutionFiles) {
-                            affectedModuleFound = true; affectedModulesFiles.add(it) }
+    SRepository repository
+
+    private static def modulePattern = ~/.*\.msd/
+    private static def modelPattern = ~/.*\.mps|.*\.model/
```
Comment (danielratiu) [inline mps-cli-gradle-plugin/plugin/src/main/groovy/org/mps_cli/cone_of_influence/Filesystem2SSolutionBridge.groovy:19]: the model patter should contain mpsr filex (root nodes saved in file per root persistency)

... at best we look if files changed inside the "models" directory ... (due to custom persistency it can be the case that we have root nodes also with other extensions) 


The layout of a directory with file-per-root persistency is "solution_name/models/model_name/(.model|rootname.mpsr|custom_persistency)"

Basically if the changes are inside a directory "models" then the model changed,

**Review by danielratiu — APPROVED** [review]
apart the identification of changed models comment the rest lgtm

thank you!

File: mps-cli-gradle-plugin/plugin/src/main/groovy/org/mps_cli/cone_of_influence/Filesystem2SSolutionBridge.groovy (hunk 1 above)
Comment (kolmar) [reply mps-cli-gradle-plugin/plugin/src/main/groovy/org/mps_cli/cone_of_influence/Filesystem2SSolutionBridge.groovy:19]: I think this case is already covered. If an `*.mpsr`/custom persistency file was modified, then we detect with `modelPattern` that there is a `.model` file in the same folder, so we know that a model was modified.
